x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/utils/event_ingested_pipeline.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract, Logger } from '@kbn/core/server';
+import { SavedObjectsClient, type ElasticsearchClient } from '@kbn/core/server';
+import { ASSET_CRITICALITY_INDEX_BASE } from '../../../../common/entity_analytics/asset_criticality';
+import type { EntityAnalyticsMigrationsParams } from '../migrations';
+import { getAllConfigurations } from '../risk_engine/utils/saved_object_configuration';
+
+export const getIngestPipelineName = (namespace: string): string => {
+  return `entity_analytics_create_eventIngest_from_timestamp-pipeline-${namespace}`;
+};
+
+export const createEventIngestedPipeline = async (
+  esClient: ElasticsearchClient,
+  namespace: string
+) => {
+  const ingestTimestampPipeline = getIngestPipelineName(namespace);
+
+  try {
+    const pipeline = {
+      id: ingestTimestampPipeline,
+      _meta: {
+        managed_by: 'entity_analytics',
+        managed: true,
+      },
+      description: 'Pipeline for adding timestamp value to event.ingested',
+      processors: [
+        {
+          set: {
+            field: 'event.ingested',
+            value: '{{_ingest.timestamp}}',
+          },
+        },
+      ],
+    };
+
+    await esClient.ingest.putPipeline(pipeline);
+  } catch (e) {
+    throw new Error(`Error creating ingest pipeline: ${e}`);
+  }
+};
+
+const getEventIngestedPipeline = async (
```

Comment:
The name suggests that it will return the pipeline instead of a boolean.
```suggestion
const hasEventIngestedPipeline = async (
```

Reply:
Good call 👍